src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -1633,6 +1625,14 @@ static void our_sofia_event_callback(nua_event_t event,
 		}
 	}
 
+	if (event == nua_r_unregister && status != 401 && status != 407 && status >= 200) {
+		switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "Mark gateway %s for destruction after unregister. Status %d.\n", sofia_private->gateway_name, status);
+		if (gateway) {
```

Comment:
Why can't we put this on the if condition above? It doesn't do anything if we don't have the gateway ptr